git/util.py

```diff
@@ -190,6 +194,10 @@ def _parse_progress_line(self, line):
         # Counting objects: 4, done.
         # Compressing objects:  50% (1/2)   \rCompressing objects: 100% (2/2)   \rCompressing objects: 100% (2/2), done.
         self._cur_line = line
+        if len(self._error_lines) > 0 or self._cur_line.startswith( ('error:', 'fatal:') ):
```

Comment:
I believe this would be a breakage in the API, as now lines that would previously be passed to update, will be caught in the base-class. Existing sub-types could depend on errors being handed to `update()`.
Do you think that makes sense, or is there another reason this has to be done in the base-class ?